resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -787,6 +796,14 @@ private[spark] class MesosClusterScheduler(
     }
   }
 
+  private def taskIsOutdated(taskId: String, state: MesosClusterSubmissionState): Boolean = {
+    if (getRetryCountFromTaskId(taskId) < getRetryCountFromTaskId(state.frameworkId)) {
```

Comment:
this can be simplified to just `getRetryCountFromTaskId(taskId) < getRetryCountFromTaskId(state.frameworkId)` without if statement